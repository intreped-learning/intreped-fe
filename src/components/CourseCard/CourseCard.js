import React from 'react';
import './CourseCard.scss';
import { Link } from 'react-router-dom';
import { addToTeacherCourses } from '../../utils/apiCalls';

const CourseCard = ({ id, category, title, description, image }) => {
  const route = `courses/${id}`

  const addToMyList = (e) => {
    e.preventDefault();
    addToTeacherCourses(id)
      .then(teacherCourse => console.log(teacherCourse));
  }

  return (
    <Link to={route}>
      <div className="coursecard">
        <img src={image} alt={title} className="thumbnail"/>
        <div className="course-info">
          <h3 className="category">{category}</h3>
          <h3 className="title">{title}</h3>
          <p className="description">{description}</p>
          <div className="card-buttons">
            <button onClick={(e) => addToMyList(e)}>Add To My List</button>
            <button>Begin Course</button>
          </div>
        </div>
      </div>
    </Link>
  );
};

export default CourseCard;