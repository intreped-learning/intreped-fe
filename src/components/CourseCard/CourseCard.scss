.coursecard {
  border: 1px solid gray;
  height: 375px;
  width: 242px;
  margin: 10px;
  color: #2B3551;
  text-align: left;
  box-shadow: 0px 5px 5px grey;
  transition: transform .2s;
  &:hover { transform: scale(1.07); }
}

.thumbnail {
  border-bottom: 2px solid #453B7F;
  // height: 180px;
  width: 240px;
}

.course-info {
  padding: 0 10px;
  height: 220px;
  width: 100%;
}

.title {
  display: inline-block;
  height: 60px;
  width: 100%;
  font-size: 1.0em;
  overflow: hidden;
}

.category {
  display: inline-block;
  font-size: .8em;
  width: 100%;
}

.description {
  display: inline-block;
  overflow-x: hidden;
  overflow-y: scroll;
  height: 60px;
  width: 100%;
  text-overflow: ellipsis;
  text-align: left;
  font-size: .8em;
}

.card-buttons {
  display: flex;
  justify-content: space-around;
  height: 30px;
  font-size: .8em;
  padding: 0;
}

.add-to-list-btn {
  background-color: #3b4a7f;
  color: #fff;
  text-decoration: none;
  transition: transform .2s;
  &:hover { font-style:italic;
    transform: scale(1.2);
    border-radius: 30px;
    text-decoration: none; }
}

.begin-course-btn {
  background-color: #8AE588;
  text-decoration: none;
  transition: transform .2s;
  &:hover { font-style:italic;
    transform: scale(1.2);
    border-radius: 30px;
    text-decoration: none; }
}