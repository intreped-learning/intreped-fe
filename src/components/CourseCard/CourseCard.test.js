import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import { BrowserRouter as Router} from 'react-router-dom';
import CourseCard from './CourseCard';

const mockStore = configureMockStore([thunk]);

describe('CourseCard', () => {
  it('should match snapshot', () => {
    const store = mockStore({
      teacher: 'Mr. Feeny',
      courses: {},
      modalOpen: true
    });
    const getWrapper = () => mount(
      <Router>
        <Provider store={store}>
          <CourseCard/>
        </Provider>
      </Router>
    );
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});