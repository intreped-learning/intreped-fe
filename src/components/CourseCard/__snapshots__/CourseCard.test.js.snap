// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <CourseCard>
        <Link
          to="courses/undefined"
        >
          <LinkAnchor
            href="/courses/undefined"
            navigate={[Function]}
          >
            <a
              href="/courses/undefined"
              onClick={[Function]}
            >
              <div
                className="coursecard"
              >
                <img
                  className="thumbnail"
                />
                <div
                  className="course-info"
                >
                  <h3
                    className="category"
                  />
                  <h3
                    className="title"
                  />
                  <p
                    className="description"
                  />
                  <div
                    className="card-buttons"
                  >
                    <button
                      onClick={[Function]}
                    >
                      Add To My List
                    </button>
                    <button>
                      Begin Course
                    </button>
                  </div>
                </div>
              </div>
            </a>
          </LinkAnchor>
        </Link>
      </CourseCard>
    </Provider>
  </Router>
</BrowserRouter>
`;
